scripts/update-gerbil-nix-recipe.ss

```diff
@@ -47,22 +50,42 @@
        ;; Thanks to vyzo for debugging this issue!!!
        (string-append "\\1\\$" new "\\2")))
 
+(def (error/dir-not-found name dir sf)
+  (define s (format "~a: directory not found\n  at: ~a" name dir))
+  (error
+    (cond
+      ((not (pair? sf))       s)
+      ((not (pair? (cdr sf))) (format "~a\n  suggested fix: ~a" s (car sf)))
+      (else                   (format "~a\n  suggested fixes: ~a" s sf)))))
+
 (def (update-recipe
       name: name
       repo: repo
       recipe-path: recipe-path
-      checkouts-dir: (checkouts-dir #f)
+      checkouts-dir: (checkouts-dir_ #f)
       source-dir: (source-dir_ #f)
       nixpkgs-dir: (nixpkgs-dir_ #f)
-      stable: (stable #f))
+      stable: (stable #f)
+      cmdopts: (cmdopts []))
+
+  (def checkouts-dir (or checkouts-dir_ (default-checkouts-dir)))
+  (unless (file-exists? checkouts-dir)
+    (error/dir-not-found "checkouts-dir" checkouts-dir ["-C" "--checkouts"]))
 
-  (def (defaultize dir reponame)
-    (path-expand (or dir reponame) (or checkouts-dir (default-checkouts-dir))))
-  (def source-dir (defaultize source-dir_ name))
-  (def nixpkgs-dir (defaultize nixpkgs-dir_ "nixpkgs"))
+  (def (defaultize dir owner reponame)
+    (cond (dir (path-expand dir checkouts-dir))
+          (else
+           (let ((d1 (path-expand reponame checkouts-dir))
+                 (d2 (path-expand reponame (path-expand owner (default-checkouts-by-owner-dir)))))
+             (if (or (file-exists? d1) (not (file-exists? d2))) d1 d2)))))
   (defvalues (repo-url site owner repo-name git-tree) (parse-repo-argument repo))
+  (def source-dir (defaultize source-dir_ owner repo-name))
+  (def nixpkgs-dir (defaultize nixpkgs-dir_ "NixOS" "nixpkgs"))
   (def recipe-file (path-expand recipe-path nixpkgs-dir))
 
+  (unless (file-exists? source-dir)
+    (error/dir-not-found name source-dir (filter (cut string-contains <> name) cmdopts)))
```

Comment:
I don't understand the string-contains logic—is that a debug tool to check that the options are well-named? That deserves a comment.

Reply:
I was thinking the `cmdopts` might contain options that apply to multiple things: the source-dir, the checkout-dir, and the nixpkgs-dir. I ended up not doing that, so I can change it back and clarify.